plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/StepCounter.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE BangPatterns     #-}
+{-# LANGUAGE DataKinds        #-}
+{-# LANGUAGE KindSignatures   #-}
+{-# LANGUAGE TypeApplications #-}
+module UntypedPlutusCore.Evaluation.Machine.Cek.StepCounter where
+
+import Control.Monad.Primitive
+import Data.Primitive qualified as P
+import Data.Proxy
+import Data.Word
+import GHC.TypeLits (Nat)
+import GHC.TypeNats (KnownNat, natVal)
+
+-- See Note [Step counter data structure]
+-- | A set of 'Word8' counters that is used in the CEK machine
+-- to count steps.
+newtype StepCounter (n :: Nat) s = StepCounter (P.MutablePrimArray s Word8)
```

Comment:
Using Word8 for the packed word-array before made sense, but does it add anything anymore with the new "real" array? Wouldn't Word64 (assuming a 64bit machine) yield the same performance or even better?

Reply:
Hmm, good question. I have no idea! I'll try it I guess.